bcgov/cas-registration
# 🚧 Breadcrumbs and dashboard architecture

Addresses [Ticket 195](https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/195)

## 🚧 Changes
    - Add client\app\components\navigation\Bread for breadcrumb navigation
    - Add client\app\components\navigation\Login for login navigation
    - Add client\app\components\navigation\Profile for user profile navigation
    - Add group route: client\app\(authenticated) for authenticated user workflows
    - Add  client\app\(authenticated)\layout.tsx for group route common layout
    - Add client\app\(authenticated)\dashboard\page.tsx for **mock** role based dashboard
    - Update client\app\operations folder to client\app\(authenticated)\dashboard\operations
    - Add group route: client\app\(onboarding) for registration workflows
    - Add  client\app\(onboarding)\layout.tsx for group route common layout
    - Update client\app\page.tsx registraion landing to client\app\(onboarding)\home\page.tsx 
    - Update client\app\middlewares\withAuthorization.tsx to route baseURL to (onboarding)\home\page.tsx to **mock auth**
    - Add client\app\data\* for app data management

## 🚀 Impact:
   - Group routes for authenticated and onboarding infrastructure (WIP) for different user roles and workflows
   - Group route (onboarding) registration page has layout: Header; Login buttons; Footer
   - Group route (authenticated) workflow pages have layout:  Header; Profile buttons; **Breadcrumbs**; Footer
   - Initial, WIP, dashboard **mocking**  different dashboards for users with different roles 

## 📝 Notes:
   - Breadcrumb component will generate dynamic breadcrumbs based on the current pathname. As the user navigates through the application, the breadcrumbs will automatically adjust to show the current path.  There may be a future need to handle specific translations for path parts that are not descriptive enough just based on the app path structure; such as, in this demo, for the "Operations\View Detail" or  "Operations\Start Registration"  page breadcrumbs which is translated from Dashboard> Operations > 4 to Dashboard> Operations > **ID** 4.


##  🔬 Testing:
🖐️
There is an expected toggle flash between display of the Login and Profile buttons on various screens which is due to the _mock_ nature of the authentication session.  

1. Open a terminal in cas-registration\bc_obps directory
2. Follow the [developer environment setup](https://github.com/bcgov/cas-registration/blob/develop/docs/developer-environment-setup.md#backend-environment-setup) to start the API
3.  Open a terminal in cas-registration\client folder
4. Start the development server:
   ```
   yarn dev
   ```
5. In your web browser, navigate to `http://localhost:3000/`
6. You should be re-directed to the registration landing page as per the _mock_ auth routing in client\app\middlewares\withAuthorization.tsx 
7. Registration landing page layout should have header, login navigation, content, and footer
![image](https://github.com/bcgov/cas-registration/assets/120038448/6c2160e0-883a-4e57-931f-1b2e9d9fba52)
8. Click login button: **"Program Administrator Log In"**
9. You should be directed to the _mock_ authenticated Admin Dashboard
10. _Mock_ authenticated Admin Dashboard page layout should have header, profile navigation,  breadcrumb navigation, content of **six** Admin dashboard tiles, footer
11. Breadcrumb navigation displays: **Dashboard** 
![image](https://github.com/bcgov/cas-registration/assets/120038448/16914aec-e37e-4e9f-a8cd-179e89cb7185)
12. Click dashboard link: "Operation Registration" tile **"Operations"**
13. You should be directed to Operations grid page
14. Operations grid page should have header, profile navigation,  breadcrumb navigation, content of operations grid, and footer
15.  Breadcrumb navigation displays: **Dashboard > Operations**
![image](https://github.com/bcgov/cas-registration/assets/120038448/6620d1ad-dc05-4a7c-9f32-141079e62bb7)
16. Click button: **"Add Operation"**
17. You should be directed to Operations form create page
18. Operations form page should have header, profile navigation,  breadcrumb navigation, content of operations form, and footer
19. Breadcrumb navigation displays: **Dashboard > Operations > Create**
![image](https://github.com/bcgov/cas-registration/assets/120038448/05075934-1e16-4d40-8d54-7dfd7a5beee9)
20. Click breadcrumb link: "Operations"
21. You should be directed to Operations grid page
22. Click a grid row action "Operations\View Detail" or  "Operations\Start Registration"  
23. You should be directed to Operations form edit page
24.  Operations form page should have header, profile navigation,  breadcrumb navigation, content of operations form, and footer
25. Breadcrumb navigation displays: **Dashboard > Operations > ID #**
![image](https://github.com/bcgov/cas-registration/assets/120038448/1b79f0db-c057-4e5e-811b-1db771e7e8c7)
26. Click header link: Log out
27. You should be directed to the registration landing page
28. Click login button: **"Industrial Operator Log In"**
29.  You should be directed to the _mock_ authenticated Operator Dashboard
30. _Mock_ authenticated Operator Dashboard page layout should have header, profile navigation,  breadcrumb navigation, content of **five** Operator dashboard tiles, footer
31. Breadcrumb navigation displays: **Dashboard** 
![image](https://github.com/bcgov/cas-registration/assets/120038448/b74f0439-9cc2-474f-9132-86518dc10052)
32. **Optional:** Repeat steps 10-25.
